spinalcordtoolbox/image.py

```diff
@@ -1850,32 +1850,36 @@ def generate_stitched_qc_images(ims_in: Sequence[Image], im_out: Image) -> Tuple
 
 def check_image_kind(img):
     """
-    Identify the image as one of 4 image types:
+    Identify the image as one of 4 image types (represented as one of 4 strings):
 
         - 'seg': Binary segmentation (0/1)
         - 'softseg': Nonbinary segmentation in the range [0, 1], where 0 and 1 are the majority of values
-        - 'labeled_seg': Nonbinary, whole values (0, 1, 2...) or (0.0, 1.0, 2.0...) only
-        - 'im': Any other float-valued image where 0 and 1 are not the majority of values.
+        - 'seg-labeled': Nonbinary, whole values (0, 1, 2...) where 0 is the majority of values
+        - 'anat': Any other image
 
     Useful for determining A) which colormap should be applied to an image, or
                            B) which interpolation can be safely used on a given image
 
     Output strings should be compatible with display_viewer_syntax() function.
     """
-    # Check if image is a segmentation (binary or soft) by making sure:
-    # - 0/1 are the two most common voxel values
-    # - 0/1 account for >95% of voxels (to allow for some soft voxels)
+    # Count the unique values in the image
     unique, counts = np.unique(np.round(img.data, decimals=1), return_counts=True)
     unique, counts = unique[np.argsort(counts)[::-1]], counts[np.argsort(counts)[::-1]]  # Sort by counts
+    # This heuristic helps to detect binary and soft segmentations
     binary_most_common = set(unique[0:2].astype(float)) == {0.0, 1.0}
     binary_percentage = np.sum(counts[0:2]) / np.sum(counts)
+    # This heuristic helps to distinguish between PSIR images and label images (2-10% zero vs. 99% zero)
     is_whole_only = np.equal(np.mod(unique, 1), 0).all()
+    zero_most_common = float(unique[0]) == 0.0
+    zero_percentage = np.sum(counts[0]) / np.sum(counts)
     if binary_most_common:
         if binary_percentage == 1.0:
             return 'seg'
         elif binary_percentage > 0.95:
             return 'softseg'
-    if is_whole_only:
+        else:  # binary_percentage <= 0.95
+            pass  # may be 'seg-labeled' or 'anat'
```

Comment:
I think this is a good way to remind future-us why the line below is `if` rather than `elif`.